File: 1/python/solution_A.py
def part1():
    sum_ = 0
    for line in open("input.txt", "r").readlines():
        digits = []
        for c in filter(lambda c: c.isnumeric(), line):
            digits.append(int(c))
        sum_ += digits[0] * 10 + digits[-1]
    return sum_


def part2():
    map_ = {
        "one": 1,
        "two": 2,
        "three": 3,
        "four": 4,
        "five": 5,
        "six": 6,
        "seven": 7,
        "eight": 8,
        "nine": 9,
    }
    sum_ = 0
    for line in open("input.txt", "r").readlines():
        digits = []
        buff = ""
        for c in line:
            buff += c
            if c.isnumeric():
                digits.append(int(c))
                buff = ""
            elif len(buff) >= 3:
                for i in range(len(buff) - 2):
                    if buff[i:] in map_:
                        digits.append(map_[buff[i:]])
        sum_ += digits[0] * 10 + digits[-1]
    return sum_


if __name__ == "__main__":
    print(part1())
    print(part2())

File: 1/go/solution_A.go
package main

import (
	"fmt"
	"os"
	"strings"
	"unicode"
)

func main() {
	fmt.Println(part1())
	fmt.Println(part2())
}

func part1() int {
	dat, err := os.ReadFile("input.txt")
	if err != nil {
		panic(err)
	}
	lines := strings.Split(string(dat), "\n")
	sum := 0
	for _, line := range lines {
		digits := []int{}
		for _, c := range line {
			if unicode.IsDigit(c) {
				digits = append(digits, int(c-'0'))
			}
		}
		sum += digits[0]*10 + digits[len(digits)-1]
	}
	return sum
}

func part2() int {
	map_ := map[string]int{
		"one":   1,
		"two":   2,
		"three": 3,
		"four":  4,
		"five":  5,
		"six":   6,
		"seven": 7,
		"eight": 8,
		"nine":  9,
	}
	dat, err := os.ReadFile("input.txt")
	if err != nil {
		panic(err)
	}
	lines := strings.Split(string(dat), "\n")
	sum := 0
	for _, line := range lines {
		digits := []int{}
		buff := ""
		for _, c := range line {
			buff += string(c)
			if unicode.IsDigit(c) {
				digits = append(digits, int(c-'0'))
				buff = ""
			} else if len(buff) >= 3 {
				for i := 0; i < len(buff)-2; i++ {
					if _, ok := map_[buff[i:]]; ok {
						digits = append(digits, map_[buff[i:]])
						buff = ""
					}
				}
			}
			fmt.Println(digits)
		}
		sum += digits[0]*10 + digits[len(digits)-1]
	}
	return sum
}
